subsys/pm/device_runtime.c

```diff
@@ -427,6 +461,12 @@ int pm_device_runtime_enable(const struct device *dev)
 		goto end;
 	}
 
+	/* lazy init of PM fields */
+	if (pm->base.dev == NULL) {
+		pm->base.dev = dev;
+		k_work_init_delayable(&pm->base.work, runtime_suspend_work);
+	}
```

Comment:
Now isr a non-isr safe devices have the same enablement code. Can we consolidate it in a single function ? You don't need to move this statement, basically just put the semphore handling inside a ISR_SAFE branch.

Reply:
I don't understand this sorry. Semaphores? And why does it need to be in a `ISR_SAFE` branch if both safe and non-safe devices behave the same?